midterm-2023-11-06/code/5029-angel-nineski/script.ts

```diff
@@ -0,0 +1,184 @@
+        if (!response.ok) {
+            throw new Error("Failed to fetch data from the JSON file");
+        }
+        const data = await response.json();
+        bands = data.metalBands;
+        populateFilterOptions(data);
+        displayBands(bands);
+    } catch (error) {
+        console.error("Error:", error);
+    }
+}
+
+// Function to display bands on the webpage
+function displayBands(bands: Band[]) {
+    const bandContainer = document.getElementById("band-container");
+    if (!bandContainer) {
+        console.error("Band container not found.");
+        return;
+    }
+
+    bandContainer.innerHTML = ''; // Clear the container
+
+    bands.forEach((band) => {
+        const bandElement = document.createElement("div");
+        bandElement.classList.add("band-table");
+
+        const location = band.location.split(", ");
+        const country = location[location.length - 1];
+
+        const albums = band.albums.map(album => album.name).join(", ");
+        const firstAlbumYear = Math.min(...band.albums.map(album => album.year));
+        const lastAlbumYear = Math.max(...band.albums.map(album => album.year));
+
+        bandElement.innerHTML = `
+            <div class="band-data">${band.id}</div>
+            <div class "band-data">${band.name}</div>
+            <div class="band-data">${band.formed}</div>
+            <div class="band-data">${location[0]}</div>
+            <div class="band-data">${country}</div>
+            <div class="band-data">${band.members.join(", ")}</div>
+            <div class="band-data">${albums}</div>
+            <div class="band-data">${firstAlbumYear}</div>
+            <div class="band-data">${lastAlbumYear}</div>
+            <div class="band-data">${firstAlbumYear} - ${lastAlbumYear}</div>
+        `;
+
+        bandContainer.appendChild(bandElement);
+    });
+
+    const headers = document.querySelectorAll(".band-header");
+    headers.forEach((header) => {
+        header.addEventListener("click", () => {
+            const sortBy = header.getAttribute("data-sort-by");
+            if (sortBy) {
+                sortTable(bands, sortBy);
+                displayBands(bands);
+            }
+        });
+    });
+}
+
+// Function to sort the table data
+function sortTable(bands: Band[], sortBy: string) {
+    bands.sort((a, b) => {
+        let aValue, bValue: number | string;
+
+        switch (sortBy) {
+            case "id":
+                aValue = a.id;
+                bValue = b.id;
+                break;
+            case "name":
+            case "location":
+            case "country":
+                aValue = a[sortBy];
+                bValue = b[sortBy];
+                break;
+            case "formed":
+            case "firstAlbumYear":
+            case "lastAlbumYear":
+                aValue = a[sortBy];
+                bValue = b[sortBy];
+                break;
+            case "yearsActive":
+                const aStartYear = a.albums[0].year;
```

Comment:
And we have the ugly part here :D BTW, you should not assume that the albums are in ascending year order.